doc/releases/release-notes-4.2.rst

```diff
@@ -116,6 +116,10 @@ New APIs and options
   * :kconfig:option:`CONFIG_SRAM_VECTOR_TABLE` moved from ``zephyr/Kconfig.zephyr`` to
     ``zephyr/arch/Kconfig`` and added dependencies to it.
 
+* Toolchain
+
+  * :c:macro:`__deprecated_version`
+
```

Comment:
This will need to go into 4.3 since we are already past the feature freeze for 4.2.